docs/dev/concepts.rst

```diff
@@ -60,93 +42,74 @@ perform. Recipes contain three important pieces of information:
+      "action": "console-log",
       "arguments": {
          "message": "It works!"
-      }
+      },
+      "filter_expression": "normandy.version >= '65.0' && stableSample(0.5)"
    }
 
-Note that the recipe as returned by the API also contains data about the action,
-including a schema describing the arguments field.
+.. info::
+
+   The data retrieved by the Normandy client is the minimum subset of data
+   that is needed execute the action. Other API end points return more
+   information about the recipe, including it's history and approval status.
 
 .. _JEXL: https://github.com/TechnologyAdvice/Jexl
 .. _Telemetry: https://wiki.mozilla.org/Telemetry
 
 .. _runtime:
 
-Runtime
--------
-The runtime is an execution environment that can affect a running instance of
-Firefox. Upon activation (typically a few moments after Firefox launches), the
-runtime:
+Normandy Client
+---------------
+The Normandy client is `a Firefox component`_ that reads and executes
+instructions from the server. Upon activation (typically every few hours),
+the runtime:
 
 1. Downloads :ref:`recipes <recipes>` from the Normandy service.
 2. Verifies the signature of the recipes.
 3. Evaluates the recipe filters and filters out recipes that do not match the
    client the runtime is installed within.
-4. Downloads the :ref:`actions <actions>` for the remaining recipes.
-5. Executes the action code for each recipe in a sandbox, passing in the
-   arguments from the recipe, and a :doc:`driver object </dev/driver>` containing
-   methods that can perform privileged actions.
-
-The runtime is implemented as `a component of Firefox in mozilla-central`_.
+4. Passes the recipes to the corresponding action code for execution.
 
-.. _a component of Firefox in mozilla-central: https://hg.mozilla.org/mozilla-central/file/tip/toolkit/components/normandy
+.. _a Firefox component: https://hg.mozilla.org/mozilla-central/file/tip/toolkit/components/normandy
 
-Threat Model
-------------
+Security Considerations
+-----------------------
 Since the goal of SHIELD is to allow Mozilla to perform certain privileged
 actions quickly without shipping full updates to Firefox, it is a tempting
 target for compromising Firefox users. Normandy includes several security
 controls to help mitigate this risk.
 
 Action and Recipe Signing
 ^^^^^^^^^^^^^^^^^^^^^^^^^
-Actions and recipes that are downloaded by the runtime are signed according to
-the Content-Signature protocol as provided by the autograph_ service. The
-runtime verifies the signature upon downloading the recipes and actions,
-ensuring that the runtime only executes recipes that have been signed with a
-Mozilla-controlled key.
+Recipes that are downloaded by the runtime are signed according to the
+Content-Signature protocol as provided by the Autograph_ service. The client
+verifies the signature upon downloading the recipes, ensuring that the
+runtime only executes recipes that have been signed with a Mozilla-controlled
```

Comment:
It feels like you're trying to remove the word `runtime` here so maybe replace it with `client`. Also, you might want to change `are signed` to `are each signed`.